docs/high-avaliability/prometheus-HA.md

```diff
@@ -1,12 +1,62 @@
+# Running Promscale itself in High-Availability mode
+
+Promscale is a stateless service, thus it can safely be run with multiple
+replicas. A load-balancer can simply route Promscale requests to any replica.
+
 # Using Promscale with Prometheus in High Availability
 
-Promscale supports running Prometheus in High Availability (HA). This means that more than one Prometheus instances can run in parallel. Promscale implements leader election to support Prometheus HA mode through pg_advisory_lock.
+Promscale supports running Prometheus in High Availability (HA). This means
+that more than you can have multiple Prometheus instances running as "HA
+pairs". Such "HA pairs" (what we'll call clusters from now on) are simply two
+or more identical Prometheus instance running on different
+machines/containers. These instances scrape the same targets and should thus
+have the same data (modulo timing of the scrapes).
+
+Promscale supports de-duplicating data coming from such HA clusters upon
+ingest. It does this by electing a leader Prometheus instance and only
+ingesting data from that leader. If the leader Prometheus stops sending data
+to Promscale for any reason, Promscale will fail-over to the other Prometheus
+instance.
+
+Promscale supports having multiple clusters sending data to the same set of
+Promscale instances. It will elect one leader per cluster.
+
+## Prometheus leader-election via external labels
+
+![Promscale architecture for Prometheus in HA using external labels](https://raw.githubusercontent.com/timescale/promscale/master/docs/high-avaliability/new_ha_system.png)
+
+To set up Promscale to process data from Prometheus running in HA mode,
+Prometheus must be configured to communicate which cluster it belongs to
+using external labels. In particular, each Prometheus should send a `cluster`
+label to indicate which cluster it's in and a `__replica__` label that
+provides a unique identifier for the Prometheus instance with the cluster.
+Two Prometheus instances running as a HA cluster, MUST send the same
+`cluster` label and different `__replica__` labels.
+
+In Kubernetes environments, it is often useful to set the `__replica__`
+label to the pod name. When using the Prometheus Operator this can be done
+with the following settings:
+
+```
+  replicaExternalLabelName: "__replica__"
+  prometheusExternalLabelName: "cluster"
```

Comment:
note according to the prom operator "cluster" is by default "prometheus" and "__replica__" is "prometheus_replica" which standard should we use?

Reply:
Is this done in Tobs or something as raw input to Prometheus operator?

Reply:
The `cluster` or `prometheus` replica will be used by the users for querying. So as `cluster` looks very generic. I prefer `prometheus` as the default label. It will give a clear understanding to the users on prometheus pair they are querying.